android/runtime/common/src/java/org/appcelerator/kroll/KrollLogging.java

```diff
@@ -119,15 +119,8 @@ public void log(String level, String... args)
 
 	private String combineLogMessages(String... args)
 	{
-		String msg;
 		int length = (args == null ? 0 : args.length);
-
-		if (length > 0) {
-			msg = args[0];
-		} else {
-			msg = new String();
-		}
-
+		String msg = (length > 0) ? args[0] : "";
```

Comment:
This feels very much like a spot we should be using `StringBuilder`

Reply:
Just optimized it now.
However, our `Ti.API` logging functions no longer call this class' variable arguments methods anymore.